src/lib/hooks/useNativeCurrency.ts

```diff
@@ -1,11 +1,9 @@
 import { NativeCurrency, Token } from '@uniswap/sdk-core'
-import { useWeb3React } from '@web3-react/core'
 import { SupportedChainId } from 'constants/chains'
 import { nativeOnChain } from 'constants/tokens'
 import { useMemo } from 'react'
 
-export default function useNativeCurrency(): NativeCurrency | Token {
-  const { chainId } = useWeb3React()
+export default function useNativeCurrency(chainId: SupportedChainId | undefined): NativeCurrency | Token {
```

Comment:
nit: shouldn't this be `chainId?: SupportedChainId` instead?

Reply:
in this case i agree